tests/test_plugin_git.py

```diff
@@ -0,0 +1,270 @@
+# Local imports
+import zambeze.orchestration.plugin_modules.git as git
+
+# Standard imports
+import os
+import pwd
+import pytest
+import random
+import socket
+
+
+@pytest.mark.gitlab_runner
+def test_git_checkCommitSuccess():
+    """Tests that the Git Plugin correctly verifies the json dict used to make
+    a commit to a file.
+
+    Test will create a file and then upload it to the GitHub Repo. The file
+    will contain a single random number.
+    """
+
+    access_token = os.getenv("ZAMBEZE84_GITHUB_ACCESS_TOKEN")
+    current_dir = os.getcwd()
+    file_name = "demofile_for_git_commit.txt"
+    f = open(current_dir + "/" + file_name, "w")
+    original_number = random.randint(0, 100000000000)
+    f.write(str(original_number))
+    f.close()
+
+    package = [
+        {
+            "commit": {
+                "repo": "TestRepo",
+                "owner": "Zambeze84",
+                "branch": "main",
+                "source": {
+                    "path": current_dir + "/" + file_name,
+                    "type": "posix absolute",
+                },
+                "destination": {"path": file_name, "type": "GitHub repository root"},
+                "commit_message": "Adding a file",
+                "credentials": {
+                    "user_name": "zambeze84",
+                    "access_token": access_token,
+                    "email": "zambeze84@gmail.com",
+                },
+            }
+        }
+    ]
+
+    git_plugin = git.Git()
+    git_plugin.configure({})
+    checked_actions = git_plugin.check(package)
+    print(checked_actions)
+    assert checked_actions["commit"][0]
+
+
+@pytest.mark.gitlab_runner
```

Comment:
can you add docstring to 1 and 2 here so we know what the difference is between the potential git failures?

Reply:
and 3 and 4. I see now it's hidden in the "# This package is..." comments